pkg/kafka/auth/config_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"testing"
+
+	"github.com/Shopify/sarama"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
+)
+
+func Test_InitFromViper(t *testing.T) {
+	v := viper.New()
+	configPrefix := "kafka.auth."
+	v.Set(configPrefix+suffixAuthentication, "tls")
+	v.Set(configPrefix+kerberosPrefix+suffixKerberosServiceName, "kafka")
+	v.Set(configPrefix+kerberosPrefix+suffixKerberosRealm, "EXAMPLE.COM")
+	v.Set(configPrefix+kerberosPrefix+suffixKerberosUseKeyTab, true)
+	v.Set(configPrefix+kerberosPrefix+suffixKerberosUsername, "user")
+	v.Set(configPrefix+kerberosPrefix+suffixKerberosPassword, "password")
+	v.Set(configPrefix+kerberosPrefix+suffixKerberosConfig, "/path/to/krb5.conf")
+	v.Set(configPrefix+kerberosPrefix+suffixKerberosKeyTab, "/path/to/keytab")
+	v.Set(configPrefix+kerberosPrefix+suffixKerberosDisablePAFXFAST, true)
+	v.Set(configPrefix+plainTextPrefix+suffixPlainTextUsername, "user")
+	v.Set(configPrefix+plainTextPrefix+suffixPlainTextPassword, "password")
+	v.Set(configPrefix+plainTextPrefix+suffixPlainTextMechanism, "SCRAM-SHA-256")
```

Comment:
We don't test flags like this. Flags should be passed as literal strings, without using variables, as an array of CLI arguments.